tests/test_checkout_integration.py

```diff
@@ -110,3 +112,115 @@ def test_checkout_flow(product_in_stock, client):
     payment = order.payments.latest('pk')
     assert payment.status == 'preauth'
     assert order.status == OrderStatus.NEW
+    latest_address = Address.objects.latest('pk')
+    assert Address.objects.are_identical(order.shipping_address, latest_address)
+
+
+def test_checkout_flow_authenticated_user(product_in_stock, client,
+                                          django_user_model):
+    """
+    Checkout with authenticated user and previously saved address
+    """
+    variant = product_in_stock.variants.get()
+    # Prepare some data
+    shipping_method = ShippingMethod.objects.create(name='DHL')
+    shipping_variant = shipping_method.price_per_country.create(price=10)
+    user = django_user_model.objects.create(
+        email='test@example.com', is_active=True)
+    user.set_password('test')
+    user.save()
+
+    shipping_data = {
+        'first_name': 'John',
+        'last_name': 'Doe',
+        'street_address_1': 'Some street',
+        'street_address_2': '',
+        'city': 'Somewhere',
+        'city_area': '',
+        'country_area': '',
+        'postal_code': '50-123',
+        'country': 'PL'}
+
+    user_shipping_address = Address.objects.create(**shipping_data)
```

Comment:
Could this be a fixture? Could this use an address known to be valid so it does not start failing at some point? `google-i18n-address` contains a lot of valid addresses in its test suite.